SYCL/ESIMD/esimd_test_utils.hpp

```diff
@@ -26,26 +26,27 @@ using namespace sycl;
 
 namespace esimd_test {
 
-// This is function provided to SYCL runtime by the application to decide
+// This is the function provided to SYCL runtime by the application to decide
 // on which device to run, or whether to run at all.
 // When selecting a device, SYCL runtime first takes (1) a selector provided by
 // the program or a default one and (2) the set of all available devices. Then
-// it passes each device to the selector. A device for which the highest number
-// is returned is selected. If a negative number was returned for all devices,
-// then the selection process will cause an exception.
+// it passes each device to the '()' operator of the selector. Device, for
+// which '()' returned the highest number, is selected. If a negative number
+// was returned for all devices, then the selection process will cause an
+// exception.
+// Require GPU device
 inline int ESIMDSelector(const device &device) {
-  if (const char *dev_filter = getenv("SYCL_DEVICE_FILTER")) {
-    std::string filter_string(dev_filter);
-    if (filter_string.find("gpu") != std::string::npos)
-      return device.is_gpu() ? 1000 : -1;
-
-    std::cerr
-        << "Supported 'SYCL_DEVICE_FILTER' env var device type is 'gpu' and "
-        << filter_string << "' does not contain that.\n";
+  std::string intel{ "Intel(R) Corporation" };
+  if (device.get_backend() == backend::ext_intel_esimd_emulator) {
+    return 1000;
+  } else if (device.is_gpu() && device.get_info<info::device::vendor>() == intel) {
```

Comment:
to avoid warnings on Linux
```suggestion
  } else if (device.is_gpu() && (device.get_info<info::device::vendor>() == intel)) {
```

Reply:
Done!